include/http.h

```diff
@@ -80,6 +82,12 @@ typedef struct http_link_context_s
     TYPE_REQUEST,
     TYPE_WEBSOCKET
   } type;
+  bool first_response;
```

Comment:
how do you mean `first_response`? response happens only once, after it is done sending data.